### PR TITLE
fix: enhance location filter options at explore page

### LINKED ISSUE #5921: Search: Fix and Enhance filter options
The filters are not consistent, e.g. it is currently not possible to choose all location types at the same time even though this is the default search behavior. Therefore please make the filter display and search functionality consistent and allow searching of past events as well. 

Please change the filters as follows and make the below options the standard search setting:
- Rename "Event Location Type" to "Setting"
  - Tick all boxes Online, Location, Mixed - by default
- If user only ticks online show all online events already even if no event name has been specified
- Date 
   - Change "All dates" filter to actually display "All dates" including past dates and move it to the end of the search filter area
   - Implement filter "Past" and "All Upcoming"
   - By default tick "All Upcoming" and when ticking "All Upcoming" tick "Today" and all other future filters (except custom dates)
   - Show "Past" at the end of the widget before "All dates"
- Categories - leave it as is to "All Categories"
- Event Types - leave it as is to "All Types"
- Ticket Type - tick Free and Paid by default
- Event Call For Speakers rename it to "Call for Speakers"
- By default tick "Open" only
- Finally change "Clear Filters" to "Default Filters" and reset filters to the standard search filter config when the user clicks it

Furthermoe
* Ensure that the search as above works in the same way from the start page as well the explore page
- The search from the start page should always use the standard search parameters for now

Screenshot from 2020-12-05 13-30-59

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #5921 

- location part is done.

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by iamareebjamal** [comment]
If all 3 location are selected = no filter

If online - is_online=true

If location - location=Something

If mixed, then is_mixed=true and location=Something

Currently, it turns others false when one is selected

**Comment by sachinchauhan2889** [comment]
@iamareebjamal sir, please review

**Comment by iamareebjamal** [comment]
Working correctly but lot of duplication

**Comment by sachinchauhan2889** [comment]
> Working correctly but lot of duplication

@iamareebjamal sir, yesterday i released that there is so much code duplication. I know how to remove it. I was about to fix at last night but i got urgent work. I will remove duplicacy by tommorow. I was seeking feedback about functionality.

**Comment by sachinchauhan2889** [comment]
@iamareebjamal sir, please review

**Comment by iamareebjamal** [comment]
There's still a lot of duplication

**Comment by sachinchauhan2889** [comment]
@iamareebjamal sir, please review. sir, if there is still code duplicacy please point that lines.

File: app/components/explore/side-bar.js [hunk 1]
```diff
@@ -16,11 +16,23 @@ export default class SideBar extends Component {
...
+  setLocationFilter() {
+    if (this.is_online && this.is_location === null && this.is_mixed === null) {
+      this.set('location', null);
+    }
+  }
 
-  @computed('category', 'sub_category', 'event_type', 'startDate', 'endDate', 'location', 'ticket_type', 'cfs', 'event_name', 'is_online', 'has_logo', 'has_image', 'is_past')
-  get hideClearFilters() {
-    return !(this.category || this.sub_category || this.event_type || this.startDate || this.endDate || this.location || this.ticket_type || this.cfs || this.event_name || this.is_online || this.has_logo || this.has_image || this.is_past);
+  setAllSettingFilter() {
```
Comment (iamareebjamal) [inline app/components/explore/side-bar.js:25]: Shouldn't it be reset

File: app/components/explore/side-bar.js [hunk 2]
```diff
@@ -67,10 +79,48 @@ export default class SideBar extends Component {
...
+      this.setProperties({
+        is_online   : 'true',
+        is_location : 'true'
+      });
+    } else if (this.is_location !== null && this.is_online !== null) {
+      this.setAllSettingFilter();
+    } else {
+      this.set('is_mixed', this.is_mixed === null ? 'true' : null);
+    }
+    this.setLocationFilter();
```
Comment (iamareebjamal) [inline app/components/explore/side-bar.js:123]: This cab be minimized, the structure is very similar

File: app/components/explore/side-bar.js [hunk 3]
```diff
@@ -178,12 +191,11 @@ export default class SideBar extends Component {
   }
 
   @action
-  clearFilters() {
+  defaultFilters() {
```
Comment (iamareebjamal) [inline app/components/explore/side-bar.js:194]: Why change name here?

**Comment by iamareebjamal** [comment]
Screenshot from 2020-12-21 06-32-30 (1)

It's not taking up the width of the column

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**